esi_leap/db/sqlalchemy/api.py

```diff
@@ -112,14 +114,20 @@ def offer_get_all(filters):
 
     query = model_query(models.Offer)
 
+    lessee_id = filters.pop('lessee_id', None)
     start = filters.pop('start_time', None)
     end = filters.pop('end_time', None)
-
     a_start = filters.pop('available_start_time', None)
     a_end = filters.pop('available_end_time', None)
 
     query = query.filter_by(**filters)
 
+    if lessee_id:
+        lessee_id_list = keystone.get_parent_project_id_tree(lessee_id)
+        query = query.filter(or_(models.Offer.project_id == lessee_id,
+                                 models.Offer.lessee_id.__eq__(None),
```

Comment:
I'm curious why lessee_id can be none when filtering by lessee_id?

Reply:
Ah, that's because if the offer has no lessee, that means that it's not limited, and anyone can claim it.

Reply:
Thanks! I see.